packages/manager/modules/sms/src/sms/batches/cancel/telecom-sms-batches-cancel.controller.js

```diff
@@ -0,0 +1,35 @@
+import get from 'lodash/get';
+
+export default class SmsBatchCancelController {
+  /* @ngInject */
+  constructor($translate) {
+    this.$translate = $translate;
+  }
+
+  executeCancelBatch() {
+    this.isCancellingBatch = true;
+
+    return this.cancelBatch(this.batch.id)
+      .then(() =>
+        this.goBack({
+          reload: true,
+        }).then(() =>
+          this.displaySuccessMessage(
+            this.$translate.instant('sms_cancel_success'),
+          ),
+        ),
+      )
+      .catch((error) =>
+        this.goBack().then(() =>
+          this.displayErrorMessage(
+            this.$translate.instant('sms_cancel_error', {
+              error: get(error, 'data.message', error.message),
+            }),
+          ),
+        ),
```

Comment:
In other places the `goBack` or `goTo{Parent}` handles the  display of the message and the reload depending on some params passed. Why not apply that pattern here ?

Reply:
done